project_wbs/models/project_project.py

```diff
@@ -0,0 +1,264 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models, _
+
+
+class Project(models.Model):
+    _inherit = "project.project"
+    _description = "WBS element"
+    _order = "c_wbs_code"
+
+    @api.multi
+    def _get_project_analytic_wbs(self):
+        result = {}
+        self.env.cr.execute('''
+            WITH RECURSIVE children AS (
+            SELECT p.id as ppid, p.id as pid, a.id, a.parent_id
+            FROM account_analytic_account a
+            INNER JOIN project_project p
+            ON a.id = p.analytic_account_id
+            WHERE p.id IN %s
+            UNION ALL
+            SELECT b.ppid as ppid, p.id as pid, a.id, a.parent_id
+            FROM account_analytic_account a
+            INNER JOIN project_project p
+            ON a.id = p.analytic_account_id
+            JOIN children b ON(a.parent_id = b.id)
+            --WHERE p.state not in ('template', 'cancelled')
+            )
+            SELECT * FROM children order by ppid
+        ''', (tuple(self.ids),))
+        res = self.env.cr.fetchall()
+        for r in res:
+            if r[0] in result:
+                result[r[0]][r[1]] = r[2]
+            else:
+                result[r[0]] = {r[1]: r[2]}
+
+        return result
+
+    @api.multi
+    def _get_project_wbs(self):
+        result = []
+        projects_data = self._get_project_analytic_wbs()
+        for ppid in projects_data.values():
+            result.extend(ppid.keys())
+        return result
+
+    @api.multi
+    @api.depends('name')
+    def name_get(self):
+        res = []
+        for project_item in self:
+            data = []
+            proj = project_item
+
+            while proj:
+                if proj and proj.name:
+                    data.insert(0, proj.name)
+                else:
+                    data.insert(0, '')
+                proj = proj.parent_id
+            data = '/'.join(data)
+            res2 = project_item.code_get()
+            if res2 and res2[0][1]:
+                data = '[' + res2[0][1] + '] ' + data
+
+            res.append((project_item.id, data))
+        return res
+
+    @api.multi
+    @api.depends('code')
+    def code_get(self):
+        res = []
+        for project_item in self:
+            data = []
+            proj = project_item
+            while proj:
+                if proj.code:
+                    data.insert(0, proj.code)
+                else:
+                    data.insert(0, '')
+
+                proj = proj.parent_id
+
+            data = '/'.join(data)
+            res.append((project_item.id, data))
+        return res
+
+    @api.multi
+    @api.depends('parent_id')
+    def _compute_child(self):
+        for project_item in self:
+            child_ids = self.search(
+                [('parent_id', '=', project_item.analytic_account_id.id)]
+            )
+            project_item.project_child_complete_ids = child_ids
```

Comment:
@aheficent IMO I think used False like `project_item.project_child_complete_ids = child_ids or False`

Reply:
if no child_ids then an empty record would be assigned. That would not trigger any error.